lib/reporters/TestReporter.js

```diff
@@ -0,0 +1,142 @@
+import {Test, Suite} from "../Data.js";
+
+var foo = new Test("foo", "group a", "passed", 1, [], false);
+var bar = new Test("bar goes wrong", "group a", "failed", 1, [{}], false);
+var baz = new Test("baz", "group b", "passed", 1, [], false);
+var skipped = new Test("skipped test", "group b", "skipped", 1, [], true);
+var subtest = new Test("subtest", "group with subgroup subgroup", "passed", 1, [], false);
+
+var groupA = new Suite("group a", [], [foo, bar]);
+var groupB = new Suite("group b", [], [baz, skipped]);
+var subgroup = new Suite("subgroup", [], [subtest]);
+var parent = new Suite("group with subgroup", [subgroup], []);
+
+// Globale Suite for QUnit
+var globalSuiteNoNesting = new Suite("", [groupA, groupB], []);
+// Globale Suite for Jasmine
+var globalSuiteNesting = new Suite("", [groupA, groupB, parent], []);
+
+
+var runGroupA = [
+    //["suiteStart", groupA],
+    //["testStart", foo],
+    ["testEnd", foo],
+    //["testStart", bar],
+    ["testEnd", bar],
+    ["suiteEnd", groupA]
+];
+
+var runGroupB = [
+    //["suiteStart", groupB],
+    //["testStart", baz],
+    ["testEnd", baz],
+    //["testStart", skipped],
+    ["testEnd", skipped],
+    ["suiteEnd", groupB]
+];
+
+var runParent = [
+    //["suiteStart", parent],
+    //["suiteStart", subgroup],
+    //["testStart", subtest],
+    ["testEnd", subtest],
+    ["suiteEnd", subgroup],
+    ["suiteEnd", parent]
+];
+
+export var testDataJasmine = [].concat(
+    //[["runStart", globalSuiteNesting]],
+    runGroupA,
+    runGroupB,
+    runParent,
+    [["runEnd", globalSuiteNesting]]
+);
+
+export var testDataQUnit = [].concat(
+    //[["runStart", globalSuiteNoNesting]],
+    runGroupA,
+    runGroupB,
+    [["runEnd", globalSuiteNoNesting]]
+);
+
+export default class TestReporter {
+    constructor(runner, referenceData) {
+        this.referenceData = referenceData.slice();
+        this.ok = undefined;
+        runner.on("runStart", this.onEvent.bind(this, "runStart"));
+        runner.on("suiteStart", this.onEvent.bind(this, "suiteStart"));
+        runner.on("testStart", this.onEvent.bind(this, "testStart"));
+        runner.on("testEnd", this.onEvent.bind(this, "testEnd"));
+        runner.on("suiteEnd", this.onEvent.bind(this, "suiteEnd"));
+        runner.on("runEnd", this.onEvent.bind(this, "runEnd"));
+    }
+
+    onEvent(event, data) {
+        var expectedEvent = this.referenceData.shift();
+
+        if (expectedEvent[0] === event) {
+            if (this.equals(expectedEvent[1], data)) {
+
+                if(this.ok === undefined && this.referenceData.length === 0) {
+                    this.ok = true;
+                }
+
+            } else {
+                this.ok = false;
+                console.error("expected:", expectedEvent[1], "actual:", data);
+            }
+        } else {
+            this.ok = false;
+            console.error("expected:", expectedEvent[0], "actual:", event);
+        }
+    }
+
+    equals(expected, actual) {
```

Comment:
This is very meta... the data structures above aren't exactly obvious either. This will take me a while to go through - maybe we can review this on a call?